lib/BoundaryValueDiffEqFIRK/src/firk.jl

```diff
@@ -730,51 +800,52 @@ end
 
 function __firk_mpoint_jacobian!(
         J, _, x, bc_diffmode, nonbc_diffmode, bc_diffcache, nonbc_diffcache, loss_bc::BC,
-        loss_collocation::C, resid_bc, resid_collocation, L::Int) where {BC, C}
-    sparse_jacobian!(@view(J[1:L, :]), bc_diffmode, bc_diffcache, loss_bc, resid_bc, x)
-    sparse_jacobian!(@view(J[(L + 1):end, :]), nonbc_diffmode,
-        nonbc_diffcache, loss_collocation, resid_collocation, x)
+        loss_collocation::C, resid_bc, resid_collocation, L::Int, p) where {BC, C}
+    DI.jacobian!(
+        loss_bc, resid_bc, @view(J[1:L, :]), bc_diffcache, bc_diffmode, x, Constant(p))
+    DI.jacobian!(loss_collocation, resid_collocation, @view(J[(L + 1):end, :]),
+        nonbc_diffcache, nonbc_diffmode, x, Constant(p))
```

Comment:
Note that if you ever work with `SparseMatrixCSC` formats (I think it is not the case here), views of sparse matrices are very inefficient